app/jobs/pdf_generator.rb

```diff
@@ -0,0 +1,103 @@
+require 'json'
+require 'net/http'
+require 'rmagick'
+
+class PdfGenerator
+    @queue = :ingests
+
+    def self.perform(pid)
+      manifest_url = self.config['vufind_test_url'].to_s + "/Item/" + pid + "/Manifest"
+      json = self.do_get manifest_url
+      parsed_json = JSON.parse json
+      large_jpegs = parsed_json["sequences"][0]["canvases"].map do |current|
+        current["images"][0]["resource"]["@id"]
+      end
+      pdf = self.generate_pdf(large_jpegs)
+      self.add_pdf_to_pid(pdf,pid)
+    end
+
+    def self.add_pdf_to_pid(pdf,pid)
+      resource = Fedora3Object.from_pid pid
+      document_list = Fedora3Object.from_next_pid
+      document_list.parent_pid = resource.pid
+      document_list.model_type = "ListCollection"
+      document_list.title = "Document List"
+      document_list.core_ingest("I")
+      document_list.collection_ingest
+      document_list.list_collection_ingest
+      image_data = self.build_document document_list, pdf, 1
+      self.add_datastreams_to_document pdf, image_data
+    end
+
+    def self.build_document(document_list, document, number)
+      document_data = Fedora3Object.from_next_pid
+      document_data.logger = @logger
+      document_data.parent_pid = document_list.pid
+      document_data.model_type = 'PDFData'
+      document_data.title = "PDF"
+      document_data.core_ingest('I')
+      document_data.data_ingest
+      document_data.document_data_ingest
+      document_data.add_sequence_relationship document_list.pid, number
+      document_data
+    end
+
+    def self.add_datastreams_to_document(document, document_data)
+      document_data.add_datastream_from_file document, 'MASTER', 'application/pdf'
+      document_data.add_master_metadata_datastream
+    end
+
+    def self.do_get(uri)
+      response = Net::HTTP.get_response(URI(uri))
+      self.check_http_error response
+      response.body
+    end
+
+    def self.check_http_error(response)
+      error = false
+      if (!response)
+        error = "No response to POST"
+      end
+      if (response && response.code[0] != "2")
+        error = "Unexpected response: #{response.code} #{response.message} #{response.body}"
+      end
+      if (error)
+        raise error
+      end
+    end
+
+    def self.config
+      Rails.application.config_for(:vudl)
+    end
+
+    def self.generate_pdf(jpegurls)
+      dir = '/tmp/'
```

Comment:
Might be worth making the /tmp directory configurable... though that's not a requirement, just an idea.